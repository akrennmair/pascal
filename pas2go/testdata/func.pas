{$mode ISO}
program test;

procedure a(function b(c : integer) : integer; i : integer);
begin
	writeln(i, ' -> ', b(i))
end;

function times2(i : integer) : integer;
begin
	times2 := i * 2
end;

function square(i : integer) : integer;
begin
	square := i * i
end;

begin
	a(times2, 23);
	a(square, 42)
end.
