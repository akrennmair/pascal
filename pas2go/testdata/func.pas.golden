package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var a func(b func(int) int, i int)
	a = func(b func(int) int, i int) {
		system.Writeln(i, " -> ", b(i))
		return
	}

	var times2 func(i int) int
	times2 = func(i int) (times2_ int) {
		times2_ = i * 2
		return
	}

	var square func(i int) int
	square = func(i int) (square_ int) {
		square_ = i * i
		return
	}

	a(times2, 23)
	a(square, 42)
}
